### PR TITLE
Add more tests around public API

### PR DESCRIPTION
**Pull request by zacharycmontoya**
- Add snapshot testing to Datadog.Trace.dll and Datadog.Trace.OpenTracing.dll on external AssemblyReferences
- Add unit test to ensure we can load "Datadog.Trace.AspNet.TracingHttpModule, Datadog.Trace.AspNet"

### CODE REVIEW

File: tracer/test/Datadog.Trace.Tests/PublicApiTests.cs [hunk 1]
```diff
@@ -11,5 +16,14 @@ public PublicApiTests()
             : base(typeof(Tracer).Assembly)
         {
         }
+
+#if NETFRAMEWORK
+        [Fact]
+        public void ExposesTracingHttpModule()
+        {
+            var httpModuleType = Type.GetType("Datadog.Trace.AspNet.TracingHttpModule, Datadog.Trace.AspNet");
```
Comment (lucaspimentel) [inline tracer/test/Datadog.Trace.Tests/PublicApiTests.cs:24]: Good one. I forgot this was a thing. And then I spent 10 minutes looking for it on nuget (it's only published in the msi).

Comment (zacharycmontoya) [reply tracer/test/Datadog.Trace.Tests/PublicApiTests.cs:24]: This is one of the things that only comes to mind once in a blue moon, so I gotta write a test while it's top of mind! 😄

**Review by pierotibou — APPROVED** [review]

File: tracer/test/Datadog.Trace.TestHelpers/PublicApiTestsBase.cs [hunk 2]
```diff
@@ -51,6 +53,26 @@ public void PublicApiHasNotChanged()
             publicApi.Should().Be(expected, "Public API should match the verified API. Update verified snapshot when the public API changes as appropriate");
         }
 
+        [Fact]
+        public void AssemblyReferencesHaveNotChanged()
+        {
+            StringBuilder sb = new();
+            foreach (var referencedAssembly in _assembly.GetReferencedAssemblies())
```
Comment (andrewlock) [inline tracer/test/Datadog.Trace.TestHelpers/PublicApiTestsBase.cs:60]: I think we should sort these by `FullName` to ensure the snaphshot is stable

**Review by andrewlock — APPROVED** [review]

File: tracer/test/Datadog.Trace.TestHelpers/PublicApiTestsBase.cs (hunk 2 above)
Comment (zacharycmontoya) [reply tracer/test/Datadog.Trace.TestHelpers/PublicApiTestsBase.cs]: Fixed in 9bec47c2be323b7046244b2d40d6be099abf15a3

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2202 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/b17a9fd4f332efd23af6b4974350f86a54297a22) will **not change** line coverage
:heavy_check_mark: Merging #2202 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/b17a9fd4f332efd23af6b4974350f86a54297a22) will **not change** branch coverage
:heavy_check_mark: Merging #2202 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/b17a9fd4f332efd23af6b4974350f86a54297a22) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/b17a9fd4f332efd23af6b4974350f86a54297a22) | #2202       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9334` / `12808` | `9321` / `12808` |          |
| Lines %   | `73%`      | `73%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4246` / `6260` | `4240` / `6260` |          |
| Branches %| `68%`      | `68%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7456`      | `7456`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84518&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84518/artifacts?artifactName=Code%20Coverage%20Report_84518&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2202](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84555&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84555/artifacts?artifactName=Code%20Coverage%20Report_84555&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/b17a9fd4f332efd23af6b4974350f86a54297a22) | #2202       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `73%`       |  `0%` :heavy_check_mark:  |
| Branches %| `68%`      | `68%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7456`      | `7456`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/e38807ccfc75f46fa7b8f085af6ce89e096db9af/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/e38807ccfc75f46fa7b8f085af6ce89e096db9af/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.CIAgentWriter](https://github.com/DataDog/dd-trace-dotnet/tree/e38807ccfc75f46fa7b8f085af6ce89e096db9af/tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs) | `-10%` :no_entry: | `-21%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/e38807ccfc75f46fa7b8f085af6ce89e096db9af/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ScopeManagerBase](https://github.com/DataDog/dd-trace-dotnet/tree/e38807ccfc75f46fa7b8f085af6ce89e096db9af/tracer/src/Datadog.Trace/ScopeManagerBase.cs) | `10%` :heavy_check_mark: | `13%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Logging.LibLogScopeEventSubscriber](https://github.com/DataDog/dd-trace-dotnet/tree/e38807ccfc75f46fa7b8f085af6ce89e096db9af/tracer/src/Datadog.Trace/Logging/LogsInjection/LibLogScopeEventSubscriber.cs) | `37%` :heavy_check_mark: | `56%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Logging.LogEnricher](https://github.com/DataDog/dd-trace-dotnet/tree/e38807ccfc75f46fa7b8f085af6ce89e096db9af/tracer/src/Datadog.Trace/Logging/LogsInjection/LogEnricher.cs) | `100%` :heavy_check_mark: | `100%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84518&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84518/artifacts?artifactName=Code%20Coverage%20Report_84518&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2202](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84555&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84555/artifacts?artifactName=Code%20Coverage%20Report_84555&api-version=6.0&%24format=zip)

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2202 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3cafe9e99979ce268515dcf5354e4454c734c2ad) will **not change** line coverage
:heavy_check_mark: Merging #2202 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3cafe9e99979ce268515dcf5354e4454c734c2ad) will **not change** branch coverage
:heavy_check_mark: Merging #2202 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/3cafe9e99979ce268515dcf5354e4454c734c2ad) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/3cafe9e99979ce268515dcf5354e4454c734c2ad) | #2202       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9218` / `12597` | `9203` / `12597` |          |
| Lines %   | `73%`      | `73%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4206` / `6160` | `4190` / `6160` |          |
| Branches %| `68%`      | `68%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7357`      | `7357`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84605&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84605/artifacts?artifactName=Code%20Coverage%20Report_84605&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2202](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84641&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84641/artifacts?artifactName=Code%20Coverage%20Report_84641&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/3cafe9e99979ce268515dcf5354e4454c734c2ad) | #2202       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `73%`       |  `0%` :heavy_check_mark:  |
| Branches %| `68%`      | `68%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7357`      | `7357`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/e7f122c64114e58f9db996f75b96006acd452bf4/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/e7f122c64114e58f9db996f75b96006acd452bf4/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.CIAgentWriter](https://github.com/DataDog/dd-trace-dotnet/tree/e7f122c64114e58f9db996f75b96006acd452bf4/tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs) | `-10%` :no_entry: | `-21%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/e7f122c64114e58f9db996f75b96006acd452bf4/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.SpanContextPropagator](https://github.com/DataDog/dd-trace-dotnet/tree/e7f122c64114e58f9db996f75b96006acd452bf4/tracer/src/Datadog.Trace/SpanContextPropagator.cs) | `7%` :heavy_check_mark: | `5%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84605&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84605/artifacts?artifactName=Code%20Coverage%20Report_84605&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2202](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84641&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84641/artifacts?artifactName=Code%20Coverage%20Report_84641&api-version=6.0&%24format=zip)

**Pull request merged**